app/models/miq_server/configuration_management.rb

```diff
@@ -2,17 +2,12 @@ module MiqServer::ConfigurationManagement
   extend ActiveSupport::Concern
   include ConfigurationManagementMixin
 
-  def get_config(type = "vmdb")
-    if is_local?
-      VMDB::Config.new(type)
-    else
-      VMDB::Config.for_resource(type, self)
-    end
+  def settings
+    (is_local? ? ::Settings : Vmdb::Settings.for_resource(self)).to_hash
```

Comment:
Actually, now that @Fryguy mentioned `#settings_for_resource`, we could use that in place of `Vmdb::Settings.for_resource(self)` here.